collections_api/handler.py

```diff
@@ -0,0 +1,110 @@
+"""This module used in a lambda and api gateway to deal with collections."""
+
+import _set_path  # noqa
+import os
+import json
+from datetime import datetime, date
+import boto3
+import re
+from pipeline_config import load_pipeline_config
+import sentry_sdk as sentry_sdk
+from sentry_sdk.integrations.aws_lambda import AwsLambdaIntegration
+from s3_helpers import read_s3_json
+
+if 'SENTRY_DSN' in os.environ:
+    sentry_sdk.init(
+        dsn=os.environ['SENTRY_DSN'],
+        integrations=[AwsLambdaIntegration()]
+    )
+
+
+def run(event, context):
+    config = load_pipeline_config(event)
+    collections_list = get_collections_list(config, event.get("source", ""))
+    if not len(collections_list):
+        return error('404')
+    return success(collections_list)
+
+
+def run_id(event, context):
+    config = load_pipeline_config(event)
+    item_json = get_id(config, event.get('id', ""))
+    if not item_json:
+        return error('404')
+    return success(item_json)
+
+
+def get_id(config: dict, id: str) -> dict:
+    """ Return content for an individual id."""
+    item_json = {}
+    key = os.path.join(config["process-bucket-data-basepath"], id + '.json')
+    print(config['process-bucket'])
+    print("key = ", key)
+    if id:
+        item_json = read_s3_json(config['process-bucket'], key)
+    return item_json
+
+
+def get_collections_list(config, source: str = "") -> list:
+    """ Get a listing of collections by source.  If no source is specified, return all."""
+    collections_list = []
+    folder = config["process-bucket-data-basepath"]
+    patterns = {
+        "aleph": "(^" + folder + r"/[0-9]{9}\.json$)",
+        "museum": "(^" + folder + r"/[A-Z]*[0-9]{4}.[.0-9]*[.a-z]*\.json$)",
+        "rbsc": "(^" + folder + r"/[A-Z]*[0-9]*_EAD\.json$)",
+        "curate": "(^" + folder + r"/[a-z0-9]{11}\.json$)"
+    }
+    s3 = boto3.client("s3")
+    paginator = s3.get_paginator("list_objects_v2")
```

Comment:
i would probaby use the s3_helpers.get_matching_s3_objects .  then it can be mocked in a test.

Reply:
I thought that only accepts a prefix and a suffix, not a regular expression.